bindings/wasm/examples/editor.html

```diff
@@ -0,0 +1,192 @@
+      line-height: 24px;
+      color: rgba(0, 0, 0, .87);
+      font-weight: 400;
+      -webkit-font-smoothing: antialiased;
+    }
+
+    p {
+      max-width: 700px;
+      margin: 1em;
+      text-align: left;
+    }
+
+    .container {
+      display: flex;
+      height: 80vh;
+    }
+
+    .col {
+      flex: 1;
+      height: 100%;
+    }
+  </style>
+</head>
+
+<body>
+  <p>This example demonstrates feeding Manifold's output into <a
+      href="https://modelviewer.dev/">&lt;model-viewer&gt;</a>
+    via the three.js GLTFExporter. The most basic three.js example is <a href="index.html">here</a>.
+  </p>
+  <button type="button" id="compile">Compile</button>
+  <div class="container">
+    <model-viewer class="col" camera-controls shadow-intensity="1" alt="Output of mesh Boolean operation"></model-viewer>
+    <div id="editor" class="col">
+    </div>
+  </div>
+</body>
+
+<script src="https://unpkg.com/three@0.144.0/build/three.js"></script>
+<script src="https://unpkg.com/three@0.144.0/examples/js/utils/BufferGeometryUtils.js"></script>
+<script src="https://unpkg.com/three@0.144.0/examples/js/exporters/GLTFExporter.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/monaco-editor/0.34.0/min/vs/loader.min.js" integrity="sha512-6bIYsGqvLpAiEBXPdRQeFf5cueeBECtAKJjIHer3BhBZNTV3WLcLA8Tm3pDfxUwTMIS+kAZwTUvJ1IrMdX8C5w==" crossorigin="anonymous" referrerpolicy="no-referrer"></script>
+<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/monaco-editor/0.34.0/min/vs/editor/editor.main.min.css" integrity="sha512-iQEIc0rsSDujsfjtD+lfyJ1W23Bh/lbgriubKDAym6VlEIDRj9rrbSIyJRyshOrl8s0yRcQ0+gyrZfSLyjJGWQ==" crossorigin="anonymous" referrerpolicy="no-referrer" />
+<script>
+  var editor = undefined;
+  var Module = {
+    onRuntimeInitialized: function () {
+      Module.setup();
+      const mesh = new THREE.Mesh(undefined, new THREE.MeshStandardMaterial({
+        color: 'yellow',
+        metalness: 1,
+        roughness: 0.2
+      }));
+      document.querySelector('#compile').onclick = function (e) {
+        monaco.languages.typescript.getTypeScriptWorker().then(worker => {
+          worker(editor.getModel().uri).then(w => {
+            w.getEmitOutput(editor.getModel().uri.toString()).then(output => {
+              const content = output.outputFiles[0].text;
+              const f = new Function('cube', 'cylinder', 'sphere', 'extrude', 'revolve',
+                'union', 'difference', 'intersection', 'THREE', 'geometry2mesh', 'show', content);
+              f(Module.cube, Module.cylinder, Module.sphere, Module.extrude,
+                Module.revolve, Module.union, Module.difference,
+                Module.intersection, THREE, 
+                geometry => new Module.Manifold(geometry2mesh(simplify(geometry))),
+                function(m) {
+                  mesh.geometry?.dispose();
+                  mesh.geometry = mesh2geometry(m.getMesh());
+                  push2MV(mesh);
+                })
```

Comment:
I've never seen this pattern before; looks cool. Can you give me a little walk-through of what it's doing?

Reply:
This function constructor is to execute strings as javascript code, with access to normal top level variables and parameters you pass into the function (names defined in the constructor and values defined when you actually call the function.

Reply:
Got it, thanks for the explanation. I see the `show` function now. Very nice!